src/plugins/minimal_scene/MinimalScene.cc

```diff
@@ -687,6 +687,10 @@ void GzRenderer::Destroy()
 
     // TODO(anyone) If that was the last scene, terminate engine?
   }
+
+  // clean up in the rendering thread
```

Comment:
I'm trying to understand why this fixes the issue. After `GzRenderer::Destroy` is called, the rendering thread exits, which means there should be no data races on `dataPtr->camera` and `dataPtr->rayQuery`, so it "should" be okay for them to be cleaned up in the destructor of `GzRenderer`. I feel like I'm missing something

Reply:
In this PR, I think these 2 calls didn't make a difference but I have a habit of doing this now whenever I see it. 

I noticed that sometimes segfaults on jammy github actions come from the destructor of rendering objects as they try to invoke calls to destroy internal ogre/ogre2 resources. If the reset is not called here, the destructors are then called when the program exits (problem with shared ptr). By that time, the engine is already destroyed and the destructor would be trying to clean up internal resources that are no longer valid causing a segfault. In ogre2, we try to detect that as much as possible but we haven't caught everything yet since those segfaults still tend to happen on github actions (jammy only). 

Two other examples are https://github.com/gazebosim/gz-sensors/pull/324 
 and https://github.com/gazebosim/gz-sim/pull/1960 where I started reseting shared ptrs and that helped fixed a few failing tests on jammy.

The bigger problem here is the excessive use of shared pointer types in gz-rendering which will probably need to be addressed at some point.

Reply:
hmm, so if destroying ogre2 resources after the engine is destroyed is problematic, is there a way to transfer ownership of those resources to the engine so it destroys them when it gets destroyed? I think Qt does that with its parent/child relationships. Maybe we can follow a similar approach. Anyways, thanks for the explanation.

Reply:
>  is there a way to transfer ownership of those resources to the engine so it destroys them when it gets destroyed?

yeah the engine actually does delete these ogre2 resources when it shuts down. In the destructor of rendering objects, we started adding checks to see if the engine is still loaded before attempting to request deletion again (by calling ogre2 destroy functions, which would cause a segfault if engine is already unloaded). I think we're just not very exhaustive in our checks. It difficult to pin point where the crash is coming from as well since it only happens on CI and not locally.